botorch/acquisition/scorebo.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+r"""
+Acquisition function for Self-Correcting Bayesian Optimization [hvarfner2023scorebo]_.
+
+References
+
+.. [Hvarfner2023scorebo]
+    C. Hvarfner, E. Hellsten, F. Hutter, L. Nardi.
+    Self-Correcting Bayesian Optimization thorugh Bayesian Active Learning.
+    In Proceedings of the Annual Conference on Neural Information
+    Processing Systems (NeurIPS), 2023.
+"""
+
+from __future__ import annotations
+
+import warnings
+
+from typing import Optional
+
+import torch
+from botorch import settings
+from botorch.acquisition.acquisition import MCSamplerMixin
+from botorch.acquisition.bayesian_active_learning import (
+    DISTANCE_METRICS,
+    FullyBayesianAcquisitionFunction,
+)
+
+from botorch.models.fully_bayesian import MCMC_DIM, SaasFullyBayesianSingleTaskGP
+
+from botorch.models.gp_regression import MIN_INFERRED_NOISE_LEVEL
+
+from botorch.models.utils import fantasize as fantasize_flag
+from botorch.sampling import MCSampler
+from botorch.sampling.normal import SobolQMCNormalSampler
+from botorch.utils.transforms import concatenate_pending_points, t_batch_mode_transform
+from torch import Tensor
+
+# The lower bound on the CDF value of the max-values
+CLAMP_LB = 1e-6
+
+
+class qSelfCorrectingBayesianOptimization(
+    FullyBayesianAcquisitionFunction, MCSamplerMixin
+):
+    def __init__(
+        self,
+        model: SaasFullyBayesianSingleTaskGP,
+        optimal_outputs: Tensor,
+        optimal_inputs: Optional[Tensor] = None,
+        X_pending: Optional[Tensor] = None,
+        distance_metric: str = "hellinger",
+        estimation_type: str = "LB",
+        sampler: Optional[MCSampler] = None,
+        num_samples: int = 64,
+        maximize: bool = True,
+    ) -> None:
+        r"""Self-correcting Bayesian optimization [hvarfner2023scorebo]_ acquisition
```

Comment:
Probably need to capitalize this to match as well
```suggestion
        r"""Self-correcting Bayesian optimization [Hvarfner2023scorebo]_ acquisition
```